### PR TITLE
fix: cache headers and isCacheMiss

### PR DESCRIPTION
**Pull request by AndreiAlexandruParaschiv**
Fix headers fetch call

### CODE REVIEW

File: assessment/libs/fetch-client.js [hunk 1]
```diff
@@ -10,108 +10,120 @@
...
+      method,
+      headers: {
+        ...options.headers,
+        Cache: 'only-if-cached',
+        'User-Agent': USER_AGENT, // always override to ensure consistency
+      },
+      ...body,
+    });
+
+    if (response.isCacheMiss) {
```
Comment (AndreiAlexandruParaschiv) [inline assessment/libs/fetch-client.js:88]: a different solution would be to use  if(!response.returnedFromCache)

File: assessment/libs/fetch-client.js [hunk 2]
```diff
@@ -10,108 +10,120 @@
...
+    const delay = (ms) => new Promise((resolve) => {
+      setTimeout(resolve, ms);
+    });
+
+    let response = await this.fetch(url, {
+      ...options,
+      method,
+      headers: {
+        ...options.headers,
+        Cache: 'only-if-cached',
```
Comment (iuliag) [inline assessment/libs/fetch-client.js:82]: How I understood the documentation is that `isCacheMiss` doesn't refer to the cache of node-fetch-cache, but rather to the actual cache on the called endpoint side.
Even with this header, if you run without having a cache locally, you'll see `isCacheMiss` is false, so it doesn't delay the next call.

File: assessment/libs/fetch-client.js (hunk 1 above)
Comment (iuliag) [reply assessment/libs/fetch-client.js:88]: Yes, this is the flag to check.

**Comment by AndreiAlexandruParaschiv** [comment]
Closing in favour of https://github.com/basecode/aem-seo-research/pull/18

**Pull request closed without merging**